src/math/karatsuba_multiplication.rs

```diff
@@ -35,8 +35,9 @@ fn _multiply(num1: i128, num2: i128) -> i128 {
 }
 
 fn normalize(mut a: String, n: usize) -> String {
-    for (counter, _) in (a.len()..n).enumerate() {
-        a.insert(counter, '0');
+    let padding = n.saturating_sub(a.len());
+    if padding > 0 {
+        a.insert_str(0, &"0".repeat(padding));
     }
```

Comment:
What do you think about:
```suggestion
    a.insert_str(0, &"0".repeat(padding));
```

Reply:
Thanks! This looks good to me, just realized the if condition isn't necessary.